bcipy/signal/process/transforms.py

```diff
@@ -0,0 +1,74 @@
+from scipy.signal import butter, filtfilt, iirnotch, sosfilt
```

Comment:
I like the composition and get_default here. Perhaps renaming to `transform`. I don't think we should collapse the filter module. We should get those concepts separate and will add things to them independently imo.

Reply:
These need docstrings and return types

Reply:
Will add doc. So you wanna have `bcipy/signal/process/transforms.py` -> `bcipy/signal/process/filter/transform.py` right?

Reply:
I'm suggesting that transforms can stay where they are. The filters themselves (bandpass, notch, downsample) would move to `bcipy/signal/process/filter.py`

Reply:
I'm gonna make the return type simpler. 
Currently it would be: `-> Union[np.ndarray, Tuple[np.ndarray, int]]`, but instead I'll just always return something for `fs` - either the updated value, or `None` if `None` was provided.

Reply:
Sorry for the confusion - I think I'm not quite picturing what you mean.
So far, I converted the filters into pytorch-style transforms - are you suggesting to take the implementation of those transforms into a separate file?

So that would be something like:
---
In `filter.py`:
```python
def downsample(data, fs, factor):
    return data[:, :: factor], fs // factor
```
In `transform.py`:
```python
class Downsample:
    ...
    def __call__(self, data, fs):
        downsample(data, fs, self.factor)
````

Reply:
In filter.py:
```python
class Bandpass:    
    def __init__(self, lo, hi, sample_rate_hz, order=5):
      nyq = 0.5 * sample_rate_hz       
      lo, hi = lo / nyq, hi / nyq 
      self.sos = butter(order, [lo, hi], analog=False, btype="band", output="sos")
    def __call__(self, data: np.ndarray, fs: Optional[int] = None): 
       if fs is None:
            return sosfilt(self.sos, data)
      return sosfilt(self.sos, data), fs

class Notch:
    ...

```
In transform.py:

```python
class Composition:    
    def __init__(self, *transforms):
        self.transforms = transforms
    def __call__(self, data: np.ndarray, fs: Optional[int] = None):
      for t in self.transforms:
         data, fs = t(data, fs)
         if fs is None:
            return data       
         return data, fs

def get_default_transform(...):
```

Downsample could probably stay in transform now that I think of it more, but I could be persuaded either way.

Reply:
Ahh ok - sounds good. 
Yea I suppose downsample isn't really a "filter" in the signal processing sense so I'll keep it in `transform.py`